### DISCUSSION TITLE
docker-compose results in NoSuchElementException

### DISCUSSION THREAD

**Discussion by lutacluny** [dadoonet/fscrawler#1586] [Q&A]
I executed the following commands: 

git clone git@github.com:dadoonet/fscrawler.git
cd fscrawler
cd contrib/docker-compose-workplacesearch
docker-compose up

But fscrawler does  not seem to work:

fscrawler-enterprisesearch-1  | [2023-01-20T10:45:04.800+00:00][7][5040][app-server][INFO]: [3af2a636-4d2e-46b6-92ef-7433d57c1035] Started HEAD "/" for 127.0.0.1 at 2023-01-20 10:45:04 +0000
fscrawler-enterprisesearch-1  | [2023-01-20T10:45:04.809+00:00][7][5040][action_controller][INFO]: Processing by SharedTogo::HomeController#index as */*
fscrawler-enterprisesearch-1  | [2023-01-20T10:45:04.811+00:00][7][5040][action_controller][INFO]: Redirected to http://localhost:3002/ws/search
fscrawler-enterprisesearch-1  | [2023-01-20T10:45:04.816+00:00][7][5040][action_controller][INFO]: Completed 302 Found in 6ms
fscrawler-enterprisesearch-1  | 127.0.0.1 - - [20/Jan/2023:10:45:04 +0000] "HEAD / HTTP/1.1" 302 0 "-" "curl/7.68.0"
fscrawler                     | Exception in thread "main" java.util.NoSuchElementException
fscrawler                     |         at java.base/java.util.Scanner.throwFor(Scanner.java:937)
fscrawler                     |         at java.base/java.util.Scanner.next(Scanner.java:1478)
fscrawler                     |         at fr.pilato.elasticsearch.crawler.fs.cli.FsCrawlerCli.main(FsCrawlerCli.java:254)

My Enviroment:

Windows 10
FSCRAWLER_VERSION=2.10-SNAPSHOT
STACK_VERSION=8.6.0
Docker version 20.10.21, build baeda1f
Docker Compose version v2.12.2

**Comment by dadoonet** [comment]
Here is the output I'm getting:

```shell
cd contrib/docker-compose-example-workplace
docker compose down -v
docker compose up
```


```
fscrawler                     | SLF4J: No SLF4J providers were found.
fscrawler                     | SLF4J: Defaulting to no-operation (NOP) logger implementation
fscrawler                     | SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
fscrawler                     | SLF4J: Class path contains SLF4J bindings targeting slf4j-api versions 1.7.x or earlier.
fscrawler                     | SLF4J: Ignoring binding found at [jar:file:/usr/share/fscrawler/lib/log4j-slf4j-impl-2.19.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
fscrawler                     | SLF4J: See https://www.slf4j.org/codes.html#ignoredBindings for an explanation.
```

**Reply by lutacluny** [reply]
In my case, the fscrawler container loops infinitly and outputs the exception as described before. What do you think have I missed here?

**Reply by dadoonet** [reply]
I'm wondering if this is happening because the behavior on Windows is different...
Which looks weird because it's running with Docker so the behavior should be similar I guess.

Just to be sure. Could you run:

```
docker compose down -v
docker compose up setup elasticsearch kibana enterprisesearch httpd
```

Then once it's ready, run:

```
docker compose up fscrawler
```

Note that I'm not using `docker-compose` but `docker compose`.

**Reply by lutacluny** [reply]
It seems to be an issue with Windows. There is no problem using WSL. `docker-compose up` works.
